src/systemdhook.c

```diff
@@ -116,21 +117,27 @@ static int bind_mount(const char *src, const char *dest, int readonly) {
 	return 0;
 }
 
-static int chcon(const char *path, const char *label) {
+static int chperm(const char *path, const char *label, int uid, int gid, bool doChown) {
 	DIR *dir;
 	struct dirent *ent;
 	if ((dir = opendir (path)) != NULL) {
 		/* print all the files and directories within directory */
 		while ((ent = readdir (dir)) != NULL) {
 			_cleanup_free_ char *full_path = NULL;
 			if (asprintf(&full_path, "%s/%s", path, ent->d_name) < 0) {
-				pr_perror("Failed to create path for chcon");
+				pr_perror("Failed to create path for chperm");
 				closedir(dir);
 				return -1;
-
 			}
 			if (setfilecon (full_path, label) < 0) {
-				pr_perror("Failed to set context %s on %s", label, path);
+				pr_perror("Failed to set context %s on %s", label, full_path);
+			}
+
+			if (doChown) {
+				/* Change uid and gid to something the container can handle */
+				if (chown(full_path, uid, gid) < 0 ) {
+					pr_perror("Failed to chown %d,%d to full_path owner: %s, errno: [%d]", uid, gid, full_path, errno);
```

Comment:
Nit:  
Failed to chown %d:%d ...

This is how you do it with the command line chown.